File: app/portfolio.py
from transaction import Transaction
class portfolio:
  transactions_ = []
  balance_ = 0.0
  shares_ = 0
  value_current_ = 0.0
  def __init__(self):
    pass

  def getEquityValue(self):
    return self.shares_*self.value_current_

  def getPortfolioValue(self):
    return self.balance_ + self.getEquityValue()

  def setCurrentValue(self, val):
    self.value_current_ = val

  def addTransaction(self, t_):
    self.transactions_.append(t_)
    if t_.transaction_ == "buy":
      self.shares_+=1
      self.balance_-=t_.value_
    elif t_.transaction_ == "sell":
      self.shares_-=1
      self.balance_+=t_.value_

File: app_cpp/app_cpp/portfolio.cpp
#include "portfolio.h"
#include <iostream>

portfolio::portfolio(double starting_balance){
  balance_ = starting_balance;
}
double portfolio::get_balance(){
  return balance_;
}
double portfolio::get_equity_value(){
  return shares_*value_current_;
}
double portfolio::get_portfolio_value(){
  return balance_+get_equity_value();
}
int portfolio::get_shares(){
  return shares_;
}
void portfolio::set_current_value(double val){
  value_current_ = val;
}
void portfolio::addTransaction(transaction t){
  transactions_.push_back(t);
  switch(t.get_type()){
    case transactionType::kBuy:
      shares_+=1;
      balance_-=t.get_value();
      break;
    case transactionType::kSell:
      shares_-=1;
      balance_+=t.get_value();
      break;
    default:
      std::cout << "unrecognized type"<<std::endl;

  }
}